hoot-core/src/main/cpp/hoot/core/conflate/tile/TileUtils.cpp

```diff
@@ -109,70 +110,31 @@ geos::geom::Envelope TileUtils::getRandomTile(
   return geos::geom::Envelope();
 }
 
-void TileUtils::writeTilesToGeoJson(
-  const std::vector<std::vector<geos::geom::Envelope>>& tiles,
+void TileUtils::writeTilesToGeoJson(const std::vector<std::vector<geos::geom::Envelope>>& tiles,
   const std::vector<std::vector<long>>& nodeCounts, const QString& outputPath,
-  const bool selectSingleRandomTile, int randomSeed)
+  const QString& fileSource, const bool selectSingleRandomTile, int randomSeed)
 {
-  // write out to temp osm and then use ogr2ogr to convert to geojson; trying to remember, but I
-  // believe our geojson writer wouldn't quite give the geometry output format needed for this,
-  // which is why ogr2ogr is used here...but that's worth verifying now.
-
-  QTemporaryFile osmTempFile("/tmp/tiles-calculate-temp-XXXXXX.osm");
-  if (!osmTempFile.open())
-  {
-    throw HootException(
-      "Unable to open OSM temp file: " + osmTempFile.fileName() + " for GeoJSON output.");
-  }
-  LOG_VARD(osmTempFile.fileName());
-  writeTilesToOsm(tiles, nodeCounts, osmTempFile.fileName(), selectSingleRandomTile, randomSeed);
+  LOG_VARD(outputPath);
 
-  // hack - To get around having to modify osmconf.ini, arbitrarily hijacking the "amenity" field
-  // here since its supported in the default ogr configuration. Doing this allows the ogr2ogr select
-  // to bring back the node counts. We can get away with this b/c there's no actual element data
-  // that could actually have an amenity tag...only tiles outlines in this file. If we decide we
-  // want to modify the default osmconf.ini as part of the build process to include a "node_count"
-  // tag, then all of this "amenity" related logic can go away. These full file  text replacements
-  // are inefficient, but the tile files aren't generally going to get huge. Ideally, our geojson
-  // I/O class could be used, but it munges the output (see above).
+  int randomTileIndex = -1;
+  if (selectSingleRandomTile)
+    randomTileIndex = TileUtils::getRandomTileIndex(tiles, randomSeed);
 
-  // replace "node_count" in the osm file with "amenity", so ogr can read it
-  FileUtils::replaceFully(
-    osmTempFile.fileName(), QStringList("node_count"), QStringList("amenity"));
+  OsmMapPtr boundaryMap = tilesToOsmMap(tiles, nodeCounts, randomTileIndex, selectSingleRandomTile);
+  boundaryMap->appendSource(fileSource);
 
-  QFile outFile(outputPath);
-  if (outFile.exists() && !outFile.remove())
-  {
-    throw HootException("Unable to open GeoJSON file: " + outputPath + " for writing.");
-  }
-  LOG_VARD(outputPath);
+  OsmGeoJsonWriter writer;
+  Settings s;
+  //  Output the source tags in the geojson writer
+  s.set(ConfigOptions::getJsonOutputTaskingManagerAoiKey(), true);
+  s.set(ConfigOptions::getWriterPrecisionKey(), 9);
+  writer.setConfiguration(s);
+  writer.open(outputPath);
+  writer.write(boundaryMap);
```

Comment:
Write everything to the new format GeoJSON without having to shell out, do text replacements, or hijacking the `amenity` tag.